File: problem 2/scripts/start_tunnel.py
import os
import sys
import subprocess
import time

PORT = int(os.getenv("PORT", "8000"))

print("=" * 65)
print("         CAMPUS ORBIT - PUBLIC HTTPS TUNNEL (PYTHON)")
print("=" * 65)
print(f"\n[*] Establishing secure public tunnel forwarding to http://127.0.0.1:{PORT}...")
print("[*] Connecting to edge tunnel network...\n")

# Try 1: Try npx localtunnel with full PATH
try:
    node_dir = r"C:\Program Files\nodejs"
    env = os.environ.copy()
    if os.path.exists(node_dir):
        env["PATH"] = f"{node_dir};" + env.get("PATH", "")
    
    npx_cmd = os.path.join(node_dir, "npx.cmd") if os.path.exists(os.path.join(node_dir, "npx.cmd")) else "npx"
    
    proc = subprocess.Popen([npx_cmd, "-y", "localtunnel", "--port", str(PORT)],
                            stdout=subprocess.PIPE, stderr=subprocess.PIPE, text=True, env=env)
    
    for line in proc.stdout:
        print(line, end="")
        if "url is:" in line:
            url = line.split("url is:")[1].strip()
            print("\n" + "=" * 65)
            print(f"🎉 LIVE PUBLIC URL GENERATED: {url}")
            print("=" * 65 + "\n")
            
    proc.wait()
except Exception as e:
    print(f"[!] Localtunnel exception: {e}")
    print(f"[*] Alternative: You can expose localhost:{PORT} using Cloudflare / Ngrok / Pinggy.")

File: problem 2/scripts/start_tunnel.js
/**
 * Campus Orbit - Live Public Tunnel Helper
 * Spawns a real HTTPS public tunnel to share with hackathon judges/evaluators.
 */
const { spawn } = require('child_process');
const fs = require('fs');
const path = require('path');

const PORT = process.env.PORT || 8000;

console.log('='.repeat(65));
console.log('         CAMPUS ORBIT - PUBLIC HTTPS TUNNEL HELPER');
console.log('='.repeat(65));
console.log(`\n[*] Establishing secure public tunnel forwarding to http://127.0.0.1:${PORT}...`);
console.log('[*] Connecting to edge tunnel network...\n');

// Ensure Node directory is in PATH
const nodeDir = 'C:\\Program Files\\nodejs';
if (fs.existsSync(nodeDir)) {
  process.env.PATH = `${nodeDir};${process.env.PATH || ''}`;
}

const tunnelProcess = spawn('npx', ['-y', 'localtunnel', '--port', PORT], {
  shell: true,
  env: process.env,
  stdio: 'pipe'
});

tunnelProcess.stdout.on('data', (data) => {
  const output = data.toString();
  console.log(output);
  if (output.includes('url is:')) {
    const url = output.split('url is:')[1]?.trim();
    console.log('\n' + '='.repeat(65));
    console.log('🎉 LIVE PUBLIC URL GENERATED:');
    console.log(`👉 ${url}`);
    console.log('Share this HTTPS link with judges, mentors, and teammates.');
    console.log('='.repeat(65) + '\n');
  }
});

tunnelProcess.stderr.on('data', (data) => {
  const err = data.toString();
  if (!err.includes('npm notice') && !err.includes('warn') && !err.includes('DeprecationWarning')) {
    console.log(`[Tunnel Info] ${err.trim()}`);
  }
});

tunnelProcess.on('close', (code) => {
  if (code !== 0) {
    console.log(`\nTunnel process ended (code ${code}). Alternative Python tunnel:`);
    console.log(`python scripts/start_tunnel.py`);
  }
});
